From 761b6159ad79cf2a92b46713e3cf7c8c347504e6 Mon Sep 17 00:00:00 2001
From: Arun Bharadwaj <arun@gumstix.com>
Date: Tue, 21 Apr 2015 16:26:34 -0700
Subject: [PATCH 02/10] u-boot breaks for the Overo boards since the following
 commit: a6b541b09022acb6f7c2754100ae26bd44eed1d9

This is because the gd pointer is not set early enough anymore,
such that the i2c_set_bus_num in get_board_revision can safely
execute. This results in a console hang at SPL and the boot does
not proceed.

This piece of code is anyway necessary only for really old Overo
boards with revision numbers <= 2410 and not required for the newer
boards. For these older boards, u-boot v2014.10 still works fine.

Signed-off-by: Arun Bharadwaj <arun@gumstix.com>
---
 board/overo/spl.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/board/overo/spl.c b/board/overo/spl.c
index 0d07149..64c33d6 100644
--- a/board/overo/spl.c
+++ b/board/overo/spl.c
@@ -29,20 +29,6 @@ int get_board_revision(void)
 {
         int revision;
 
-#ifdef CONFIG_SYS_I2C_OMAP34XX
-        unsigned char data;
-
-        /* board revisions <= R2410 connect 4030 irq_1 to gpio112             */
-        /* these boards should return a revision number of 0                  */
-        /* the code below forces a 4030 RTC irq to ensure that gpio112 is low */
-        i2c_set_bus_num(TWL4030_I2C_BUS);
-        data = 0x01;
-        i2c_write(0x4B, 0x29, 1, &data, 1);
-        data = 0x0c;
-        i2c_write(0x4B, 0x2b, 1, &data, 1);
-        i2c_read(0x4B, 0x2a, 1, &data, 1);
-#endif
-
         if (!gpio_request(112, "") &&
             !gpio_request(113, "") &&
             !gpio_request(115, "")) {
-- 
2.1.4

